sklearn/metrics/_classification.py

```diff
@@ -274,6 +274,9 @@ def confusion_matrix(y_true, y_pred, labels=None, sample_weight=None,
 
     if labels is None:
         labels = unique_labels(y_true, y_pred)
+    elif len(y_true) == 0:
```

Comment:
Since we are checking the array, I think this is more pythonic to do the following to check for emptyness.

```suggestion
    elif not y_true:
```

Reply:
Ok, so it looks like after calling `_check_targets` and then immediately raising an error if `y_type` is not binary or multiclass we know that `y_true` and `y_pred` must be 1d numpy arrays.  I don't think that numpy arrays are truthy and falsey like lists are.
```python
import numpy as np

print(bool(np.array([])))
# prints false but raises a deprecation warning 
# suggesting `array.size > 0` instead
# in the future an error will be returned

print(bool(np.array([0])))
# prints False

print(bool(np.array([1])))
# prints True

print(bool(np.array([0, 0])))
# raises an error that the truth value of an array 
# with more than one element is ambiguous
```

So it sounds like the suggestion from numpy is to use `elif y_true.size == 0`.

Reply:
yep I thought that it was behaving as a list. so let's keep your code, this is the best solution I think